regen-network/regen-web
# Project metadata form

## Description

Closes: regen-network/regen-registry#1416

- format JSON input with indentation
- adds metadata to project /edit
- re-uses logic from https://github.com/regen-network/regen-web/pull/1746 to submit metadata, so now supporting on chain project metadata update
- adds/updates some project metadata/forms related types



---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. Go to https://deploy-preview-1782--regen-registry.netlify.app/project-pages/C01-001/edit/metadata
2. Connect to Keplr using our registry test account `regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46` which is the admin for this project
3. Update some fields in the metadata that are visible on the project page "additional info" section so you can see those changes there once submitted: https://deploy-preview-1782--regen-registry.netlify.app/project/C01-001, eg `regen:projectType`

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
